lib/stemcell/metadata_source.rb

```diff
@@ -55,6 +55,15 @@ def expand_role(role, environment, override_options={}, options={})
 
       raise EmptyRoleError if !allow_empty_roles && role_empty
 
+      # Step 1.5: Override context specific values
+      if !role_empty
+        context_overrides = role_options['context_overrides'] || {}
+        override_contexts.each { |context|
+          overriding_hash = context_overrides[context]
+          role_options.merge!(overriding_hash) if overriding_hash
+        }
```

Comment:
👌  we probably should emphasize the `context_overrides` in Chef is reserved for a special use.